cat_merge/file_utils.py

```diff
@@ -47,6 +47,45 @@ def write_tar(tar_path: str, files: List[str], delete_files=True):
             os.remove(file)
 
 
+def read_tar_dfs(tar: tarfile.TarFile, type_name, add_provided_by: bool = True) -> List[pd.DataFrame]:
+    dataframes = []
+    for member in tar.getmembers():
+        if member.isfile() and type_name in member.name:
+            dataframes.append(read_tar_df(tar.extractfile(member), provided_by = member.name))
+    return dataframes
+
+
+def read_tar_df(file: tarfile.TarInfo, provided_by: str = None, add_provided_by: bool = True) -> pd.DataFrame:
+    df = pd.read_csv(file, sep="\t", dtype="string", lineterminator="\n", quoting=csv.QUOTE_NONE, comment='#')
+
+    if provided_by != None:
```

Comment:
We wouldn't want to actually set provided_by (which stores which ingest output file a node or edge comes from) when reading the finished kg, so it might make more sense to remove the provided_by parts. (Could also set add_provided_by to default to false)

Reply:
I had a similar thought with the provided_by features. My inclination is to have them off by default and to set a different value, i.e. kg_import_from. Then if we need to know how the archive was read rather than how the merge was performed we can have both pieces of information. Does that make sense?